### PR TITLE
Split file into three: cli, lib, and rpc flavor

### PR DESCRIPTION
**Pull request by altano**
Split file so that it can be required as a module (lib), called from the cli (node ./prettier-cli.js file1.js file2.js), and started as an rpc server (node ./prettier-rpc.js).

### CODE REVIEW

File: prettier-cli.js [hunk 1]
```diff
@@ -0,0 +1 @@
+module.exports = require('prettier/bin/prettier.js');
```
Comment (vjeux) [inline prettier-cli.js]: nit: missing \n at the end of the file

File: prettier-lib.js [hunk 2]
```diff
@@ -0,0 +1 @@
+module.exports = require('prettier');
```
Comment (vjeux) [inline prettier-lib.js]: same here

File: rollup.config.js [hunk 3]
```diff
@@ -1,16 +1,39 @@
-import resolve from 'rollup-plugin-node-resolve';
 import commonjs from 'rollup-plugin-commonjs';
 import json from 'rollup-plugin-json';
+import replace from 'rollup-plugin-replace';
+import resolve from 'rollup-plugin-node-resolve';
+
+const version = 
+  process.env.rpc ? 'rpc' :
```
Comment (vjeux) [inline rollup.config.js:7]: Nice, in my attempt, I copy pasted this file three times...

File: rollup.config.js [hunk 4]
```diff
@@ -1,16 +1,39 @@
...
 
 export default {
-  entry: 'prettier-rpc.js',
-  dest: 'dist/prettier-rpc-' + require('./package.json').dependencies.prettier + '.min.js',
+  entry: `prettier-${version}.js`,
+  dest: `dist/prettier-${version}-${require('./package.json').dependencies.prettier}.min.js`,
   format: 'cjs',
+  banner: version === 'cli' ? '#!/usr/bin/env node\n' : undefined,
   plugins: [
+    replace({
```
Comment (vjeux) [inline rollup.config.js:21]: Nice, I made a hacky script to do that :( So much better!

File: rollup.config.js [hunk 5]
```diff
@@ -1,16 +1,39 @@
...
+      '#!/usr/bin/env node': '',
```
Comment (vjeux) [inline rollup.config.js]: you probably want to include the \n, otherwise it's going to have one too many (not a big deal)

**Review by vjeux — APPROVED** [review]

File: rollup.config.js (hunk 5 above)
Comment (altano) [reply rollup.config.js]: You *should* be right but apparently it makes no difference in the output as the extra newline is already removed. Rollup must have some logic around this. I'm putting in the \n anyway though.

File: prettier-lib.js (hunk 2 above)
Comment (altano) [reply prettier-lib.js]: Fixed

File: prettier-cli.js (hunk 1 above)
Comment (altano) [reply prettier-cli.js]: Fixed

**Pull request merged**